### ISSUE TITLE
`ansible-base` should be switched to `ansible-core`

### ISSUE THREAD

**Issue by gotmax23** [ansible-community/molecule#3285]
Hi Molecule developers,

## Issue Type

- Bug report

## Molecule and Ansible details

``` console
$ ansible --version && molecule --version
ansible [core 2.11.6] 
  config file = /home/gotmax/.ansible.cfg
  configured module search path = ['/home/gotmax/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.10/site-packages/ansible
  ansible collection location = /home/gotmax/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.10.0 (default, Oct  4 2021, 00:00:00) [GCC 11.2.1 20210728 (Red Hat 11.2.1-1)]
  jinja version = 3.0.2
  libyaml = True
molecule 3.5.2 using python 3.10 
    ansible:2.11.6
    delegated:3.5.2 from molecule
    docker:1.1.0 from molecule_docker requiring collections: community.docker>=1.9.1
    podman:1.0.1 from molecule_podman requiring collections: containers.podman>=1.7.0 ansible.posix>=1.3.0
```

Molecule installation method (one of):

- pip

Ansible installation method (one of):

- OS package

[Non applicable questions removed]

## Description

I think we should switch the `ansible-base` extra dependency to `ansible-core` (or at least add another extra called `ansible-core`). After `ansible-base` 2.10, the name of the minimal ansible package was changed to `ansible-core`. Currently, `pip install molecule[ansible] --user` installs ansible (the community package that just includes collections) and its dependency `ansible-core<2.12,>=2.11.6` (the new version of the minimal Ansible package), but `pip install molecule[ansible-base] --user` installs a different version of the minimal Ansible package.

I know that `ansible` is an optional dependency and it is possible to just run `pip3 install molecule ansible-core`, but I still think we should switch it to avoid user confusion and prevent conflicting dependencies. What do you think?

Related: https://github.com/thegeeklab/ansible-later/issues/242

Thanks,
Maxwell

**Comment by mafalb** [comment]
It is valid to want ansible-base installed (if you want to test explicitly against 2.10), so I would I would suggest not to switch but add another extra for ansible-core, then one could use either molecule[ansible] xor molecule[ansible-base] xor molecule[ansible-core].

```
ansible =
    ansible >= 2.10
ansible-base =
    ansible-base ~= 2.10.0
ansible-core =
    ansible-core >= 2.11.0
```

**Comment by gotmax23** [comment]
It looks like this was fixed in #3295. I'm not sure what the rationale was for completely removing all the ansible extras, though.
